webapp/courses/templates/courses/base.html

```diff
@@ -163,19 +164,29 @@
       {% endblock %}
 
       {% block termbank %}
-        {% if terms %}
+        {% if termbank_contents %}
           <div class="termbank" id="termbank">
             <button class="expand-box" onclick="expand_box(this);" title="{% trans 'Hide/Show the termbank' %}">◀</button>
             <div class="termbank-contents" id="termbank-contents">
               <div class="termbank-heading" id="termbank-heading">{% trans "Termbank" %}</div>
+              <div class="termbank-search-div">
+                <span>{% trans "Search:" %}</span>
+                <input id="termbank-search" type="search" oninput="filter_termbank_contents($(this).val());">
+              </div>
               <div id="termbank-list-div" class="list-div">
-                <ul>
-                  {% for term in terms %}
-                    <li class="term-list-item">
-                      <span class="term" id="{{ term.span_id }}" onmouseover="show_term_description_during_hover(this, '#{{ term.slug }}-term-div');">{{ term.name }}</span>
-                    </li>
+                <ol>
+                  {% for title_letter, terms_by_letter in termbank_contents %}
+                    <li class="terms-by-letter">
+                      <span class="term-title-letter">{{ title_letter }}</span>
+                      <ul>
```

Comment:
Now that I think of it, these should be ol, too, since the terms _are_ in alphabetical order.